src/material/dialog/dialog.ts

```diff
@@ -144,13 +146,21 @@ export abstract class _MatDialogBase<C extends _MatDialogContainerBase> implemen
       (typeof ngDevMode === 'undefined' || ngDevMode)) {
       throw Error(`Dialog with id "${config.id}" exists already. The dialog id must be unique.`);
     }
+    if (this.openDialogs.length && componentOrTemplateRef === this._lastDialogOpenType &&
+        this._shouldBlockAnotherDialogOpening()) {
+      throw Error(`Ignoring attempt to open dialog${
+          config.id ? ' with id "' + config.id + '"' :
+                      ''} since a dialog of the same type was opened within the last 300ms.`);
```

Comment:
We can try presubmitting it, but I suspect that throwing an error will be super breaking for tests that haven't been flushing their animations.

Reply:
What would you suggest as a better approach?

Reply:
Didn't we talk about returning the previous dialog ref if the user opens a new dialog with the same component before the animation is done?

Reply:
That wasn't mentioned. You did have [some concerns](https://github.com/angular/components/pull/23085#issuecomment-871568191) about throwing an error, but @jelbourn only [responded with](https://github.com/angular/components/pull/23085#issuecomment-871635718):

> Rather than throwing an error for this case, we could try to figure out if we can decide not to care as much about that case, since multiple/nested modal dialogs are already somewhat of an anti-pattern.

I wasn't sure what approach this was suggesting.

Reply:
OK, this has been updated to return the previous `MatDialogRef` instead of throwing.